app/models/potion.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class Potion < ApplicationRecord
+  belongs_to :game
+  belongs_to :canonical_potion, optional: true, class_name: 'Canonical::Potion'
+
+  has_many :potions_alchemical_properties, dependent: :destroy, inverse_of: :potion
+  has_many :alchemical_properties, through: :potions_alchemical_properties
+
+  validates :name, presence: true
+  validates :unit_weight, numericality: { greater_than_or_equal_to: 0, allow_blank: true }
+
+  before_validation :set_canonical_potion
+
+  def canonical_models
+    return Canonical::Potion.where(id: canonical_potion_id) if canonical_potion.present?
```

Comment:
The goal of this is to ensure that an `ActiveRecord::Relation` object is always returned from this method and not a single `Canonical::Potion` model. This simplifies the API.